src/Nethermind/Nethermind.Trie/RustVerkleLib.cs

```diff
@@ -209,6 +212,17 @@ public static unsafe void VerkleTrieInsert(RustVerkle verkleTrie, byte[] key, by
         
     }
     
+    public static unsafe byte[] CalculatePedersenHash(Span<byte> value)
+    {
+        fixed (byte* p = &MemoryMarshal.GetReference(value))
+        {
+            IntPtr hash = calculate_pedersan_hash(p);
+            byte[] managedValue = new byte[32];
+            Marshal.Copy(hash, managedValue, 0, 32);
+            return managedValue;
```

Comment:
Not sure what is later done with it, but if its just temporary value we can use this pattern and avoid allocation:

https://github.com/NethermindEth/rocksdb-sharp/blob/master/RocksDbSharp/Native.Marshaled.cs#L160-L188

if its saved in state somewhere probably better leave it as is.